.github/workflows/buildserver.yml

```diff
@@ -0,0 +1,83 @@
+name: publish a docker image
+
+on:
+  push:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - cuda_version: "12.6.2"
+            ubuntu_version: "24.04"
+            dockerfile: "Dockerfile.client"
+            image_suffix: "-client"
+          - cuda_version: "12.6.2"
+            ubuntu_version: "24.04"
+            dockerfile: "Dockerfile.server"
+            image_suffix: "-server"
+          - cuda_version: "12.5.1"
+            ubuntu_version: "22.04"
```

Comment:
Do different Ubuntu versions result in notable differences in build?

Reply:
> the container version MUST match your local cuda version, so supporting multiple versions is needed.

Yes this is correct. Thanks for the context, sounds good. I didn't know that older CUDA versions aren't on any newer Ubuntu versions. That is unfortunate indeed...